site/public/js/drag-and-drop.js

```diff
@@ -672,7 +672,8 @@ function handleSubmission(days_late, late_days_allowed, versions_used, versions_
                     alert("ERROR! You may not use angle brackets in your filename: " + file_array[i][j].name);
                     return;
                 }
-                formData.append('files' + (i + 1) + '[]', file_array[i][j]);
+                formData.append('files' + (i + 1) + '[]', file_array[i][j], encodeURIComponent(file_array[i][j].name)
```

Comment:
Did you mean to leave the 3rd argument here?

Reply:
Yes I did, the third argument was a bit long so I wrote it over 2 lines, but yea files should be encoded there.
```
formData.append('files' + (i + 1) + '[]', file_array[i][j], encodeURIComponent(file_array[i][j].name)
          .replace(/\!/g, "%21").replace(/\(/g, "%28").replace(/\)/g, "%29"));
```

Reply:
Oh sorry I think I misread where that code is, so I think your right with the third argument, there should be a different method to encode bulk uploads since it looks like that is based on timestamps instead

Reply:
Encoding methods for both handle submission and handle bulk submission are now updated and further tested.